robots/pr-labeler/main.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Commenter provides a way to --query for issues and append a --comment to matches.
+//
+// The --token determines who interacts with github.
+// By default commenter runs in dry mode, add --confirm to make it leave comments.
+// The --updated, --include-closed, --ceiling options provide minor safeguards
+// around leaving excessive comments.
+package main
+
+import (
+	"flag"
+	"log"
+	"net/url"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+)
+
+const (
+	needsOkToTest = "needs-ok-to-test"
+	okToTest      = "ok-to-test"
+)
+
+type client interface {
+	AddLabel(org, repo string, number int, label string) error
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	GetPullRequests(org, repo string) ([]github.PullRequest, error)
+	ListCollaborators(org, repo string) ([]github.User, error)
+	ListOrgMembers(org, role string) ([]github.TeamMember, error)
+}
+
+type options struct {
+	confirm  bool
+	endpoint flagutil.Strings
+	org      string
+	repo     string
+	token    string
+}
+
+func flagOptions() options {
+	o := options{
+		endpoint: flagutil.NewStrings("https://api.github.com"),
+		org:      "kubernetes",
+		repo:     "test-infra",
+	}
+	flag.BoolVar(&o.confirm, "confirm", false, "Mutate github if set")
+	flag.StringVar(&o.token, "token", "", "Path to github token")
```

Comment:
consider: technically I think --token-path is slightly more common (although I prefer --token)